tls/s2n_connection.c

```diff
@@ -352,6 +352,19 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
         conn->multirecord_send = true;
     }
 
+    conn->status_request_type = config->status_request_type;
+
+    /* status_request_type defaults to S2N_STATUS_REQUEST_NONE, and is set by users via
+     * s2n_config_set_status_request_type. However, status_request_type can also be set
+     * via s2n_config_set_verification_ca_location. To ensure backwards compatibility,
+     * this behavior is kept for clients. For servers, however, this value is reset to
+     * S2N_STATUS_REQUEST_NONE if the user did not intentionally set this value in
+     * s2n_config_set_status_request_type.
+     */
+    if (!config->status_request_type_set && conn->mode == S2N_SERVER) {
+        conn->status_request_type = S2N_STATUS_REQUEST_NONE;
+    }
```

Comment:
We might want to keep the behavior where status_request_type is set to S2N_STATUS_REQUEST_NONE if s2n_x509_ocsp_stapling_supported() is false. What currently happens if we try to set ocsp stapling when it's not supported?

Reply:
`s2n_config_set_status_request_type` errors if a user tries to set the status request type to `S2N_STATUS_REQUEST_OCSP` and OCSP stapling isn't supported. So I don't think it's possible for `status_request_type` to ever be `S2N_STATUS_REQUEST_OCSP` if it's not supported.